lib/kubernetes-deploy/runner.rb

```diff
@@ -293,37 +283,36 @@ def validate_configuration
       end
 
       raise FatalDeploymentError, "Configuration invalid: #{errors.join(', ')}" unless errors.empty?
-      KubernetesDeploy.logger.info("All required parameters and files are present")
+      @logger.info("All required parameters and files are present")
     end
 
     def deploy_resources(resources, prune: false)
-      KubernetesDeploy.logger.info("Deploying resources:")
+      @logger.info("Deploying resources:")
 
       # Apply can be done in one large batch, the rest have to be done individually
       applyables, individuals = resources.partition { |r| r.deploy_method == :apply }
 
       individuals.each do |r|
-        KubernetesDeploy.logger.info("- #{r.id}")
+        @logger.info("- #{r.id}")
         r.deploy_started = Time.now.utc
         case r.deploy_method
         when :replace
-          _, _, st = run_kubectl("replace", "-f", r.file.path)
+          _, _, st = kubectl.run("replace", "-f", r.file.path, log_failure: false)
         when :replace_force
-          _, _, st = run_kubectl("replace", "--force", "-f", r.file.path)
+          _, _, st = kubectl.run("replace", "--force", "-f", r.file.path, log_failure: false)
         else
           # Fail Fast! This is a programmer mistake.
           raise ArgumentError, "Unexpected deploy method! (#{r.deploy_method.inspect})"
         end
 
+        next if st.success?
+        # it doesn't exist so we can't replace it
+        _, err, st = kubectl.run("create", "-f", r.file.path, log_failure: false)
         unless st.success?
```

Comment:
This is a new `st` from line 310: the status of the create command, rather than the original replace command.